File: sdks/flutter/vpay_checkout_flutter/test/redaction_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:vpay_checkout_flutter/src/redaction.dart';

void main() {
  group('redacted', () {
    test('renders the exact character count, not the value', () {
      expect(redacted('pi_123_secret_abcdefgh'), '[22 chars redacted]');
      expect(redacted('pi_123_secret_abcdefgh'), isNot(contains('pi_123')));
      expect(redacted('pi_123_secret_abcdefgh'), isNot(contains('abcdefgh')));
    });

    test(
      'renders the literal null for a null value, not [0 chars redacted]',
      () {
        expect(redacted(null), 'null');
      },
    );

    test('renders a zero-length secret distinctly from null', () {
      expect(redacted(''), '[0 chars redacted]');
      expect(redacted(''), isNot('null'));
    });
  });
}

File: sdks/tauri/tauri-plugin-vpay-checkout/guest-js/redaction.test.ts
import { describe, expect, it } from "vitest";

import { redacted } from "./redaction.js";

describe("redacted", () => {
  it("renders the exact character count, not the value", () => {
    expect(redacted("cs_123_secret_abcdef")).toBe("[20 chars redacted]");
    expect(redacted("cs_123_secret_abcdef")).not.toContain("secret");
  });

  it("renders the literal null for a null value, not [0 chars redacted]", () => {
    expect(redacted(null)).toBe("null");
  });

  it("renders undefined the same way it renders null", () => {
    expect(redacted(undefined)).toBe("null");
  });

  it("renders a zero-length secret distinctly from null", () => {
    expect(redacted("")).toBe("[0 chars redacted]");
    expect(redacted("")).not.toBe(redacted(null));
  });
});
